pages/campaigns/donations.tsx

```diff
@@ -0,0 +1,88 @@
+import Box from '@mui/material/Box'
+import Button from '@mui/material/Button'
+import Typography from '@mui/material/Typography'
+import { DataGrid } from '@mui/x-data-grid'
+import DocumentListTabs from 'components/DocumentListTabs'
+import Searchbar from 'components/Searchbar'
+import { donationFilters } from 'components/Searchbar/defaults'
+import { DocumentStatus } from 'constants/DocumentStatus'
+import { CampaignStatus } from 'constants/Donation'
+import type { NextPage } from 'next'
+import { useEffect, useState } from 'react'
+import { Campaign, campaignColumns } from 'models/campaign'
+
+const testData: Campaign[] = [
+  {
+    id: 1,
+    name: 'Impact Fund',
+    details: 'Details of Impact Fund go here',
+    donors: 15,
+    amount: 15,
+    country: 'Singapore',
+    status: CampaignStatus.PUBLISHED,
+  },
+  {
+    id: 2,
+    name: 'Impact Fund v2',
+    details: 'Details of Impact Fund v2 go here',
+    donors: 10,
+    amount: 200,
+    country: 'Singapore',
+    status: CampaignStatus.ARCHIVED,
+  },
+  {
+    id: 3,
+    name: 'Impact Fund v3',
+    details: 'Details of Impact Fund v3 go here',
+    donors: 19,
+    amount: 50000,
+    country: 'Singapore',
+    status: CampaignStatus.DRAFT,
+  },
+]
+
+const CampaignList: NextPage = () => {
+  // TODO get this from a API call
+  const tags = ['food', 'clothes', 'mental-health', 'financial', 'training']
+
+  const [campaigns, setCampaigns] = useState<Campaign[]>([])
+  const [search, setSearch] = useState<string>('')
+  const [status, setStatus] = useState<DocumentStatus>(DocumentStatus.All)
+  const { props: _props, filters } = donationFilters(tags)
+
+  useEffect(() => {
+    const props = { ..._props, status }
+    setCampaigns(testData)
+    // setCampaigns(someApi(search, props))
+  }, [_props, status, search])
+
+  const handleView = (row: Campaign) => {
+    // route(`/donation/view/${row.id}`, row)
+  }
+
+  return (
+    <Box sx={{ display: 'flex', flexDirection: 'column' }}>
+      <Typography sx={{ marginTop: 2, marginBottom: 2 }} variant="h4">
+        Donations
+      </Typography>
+      <Box sx={{ display: 'flex', justifyContent: 'space-between' }}>
+        <Button color="secondary" variant="contained">
+          Create New Campaign
+        </Button>
+        <Searchbar search={search} setSearch={setSearch} filters={filters} />
+      </Box>
+      <Box sx={{ flex: 1 }}>
+        <DocumentListTabs status={status} setStatus={setStatus}>
+          <DataGrid
+            columns={campaignColumns(handleView)}
+            rows={campaigns}
+            autoHeight
+            pageSize={10}
+          />
```

Comment:
can think about how you would display the data for the diff tabs

Reply:
I thought it was to just use the tabs as a filter as such: https://github.com/dsc-project-mr/mr-fe-internal/blob/19337966cd8b23844e29e02522d71d7834984314/pages/campaigns/donations.tsx#L53-L57

I'm assuming the filtering is done server-side but we provide the prop filters.

Reply:
ohh i was thinking like if we do it server side should we still like hold the data for the diff tabs too? so we dont have to keep calling the apis everytime we switch tabs